tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/TraceAnnotations/TraceAnnotationInfoFactory.cs

```diff
@@ -24,14 +24,15 @@ public static TraceAnnotationInfo Create(MethodBase? method)
             }
             else
             {
+                string defaultResourceName = method.DeclaringType is null ? method.Name : method.DeclaringType!.Name + "." + method.Name;
```

Comment:
It's weird the `!` is required here 🤔 Is it required if you invert it?
```suggestion
                string defaultResourceName = method.DeclaringType is not null? method.DeclaringType.Name + "." + method.Name : method.Name;
```

Reply:
It turns out the `!` is not actually needed, I just reflexively added it 😆 I need to rebase this work anyways so I'll make a small change to remove it

Reply:
Fixed in 73844dfb8cd60e9353407fe04e2726122fb76be7